WordPress/src/main/java/org/wordpress/android/ui/notifications/MilestoneDetailFragment.kt

```diff
@@ -0,0 +1,220 @@
+    private var noteBlockAdapter: NoteBlockAdapter? = null
+
+    @Inject
+    lateinit var imageManager: ImageManager
+
+    @Inject
+    lateinit var notificationsUtilsWrapper: NotificationsUtilsWrapper
+
+    @Inject
+    @Named(IO_THREAD)
+    lateinit var ioDispatcher: CoroutineDispatcher
+
+    @Inject
+    @Named(UI_THREAD)
+    lateinit var mainDispatcher: CoroutineDispatcher
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        (requireActivity().application as WordPress).component().inject(this)
+        if (savedInstanceState != null && savedInstanceState.containsKey(KEY_NOTE_ID)) {
+            restoredNoteId = savedInstanceState.getString(KEY_NOTE_ID)
+            restoredListPosition = savedInstanceState.getInt(KEY_LIST_POSITION, 0)
+        }
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        notification?.let {
+            outState.putString(KEY_NOTE_ID, it.id)
+            outState.putInt(KEY_LIST_POSITION, listView.firstVisiblePosition)
+        } ?: run {
+            // This is done so the fragments pre-loaded by the view pager can store the already rescued restoredNoteId
+            if (!TextUtils.isEmpty(restoredNoteId)) {
+                outState.putString(KEY_NOTE_ID, restoredNoteId)
+            }
+        }
+
+        super.onSaveInstanceState(outState)
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        val view = inflater.inflate(R.layout.notifications_fragment_detail_list, container, false)
+        rootLayout = view.findViewById(R.id.notifications_list_root)
+        return view
+    }
+
+    @Suppress("DEPRECATION", "OVERRIDE_DEPRECATION")
+    override fun onActivityCreated(bundle: Bundle?) {
+        super.onActivityCreated(bundle)
+        val listView = listView
+        listView.divider = null
+        listView.dividerHeight = 0
+        listView.setHeaderDividersEnabled(false)
+        reloadNoteBlocks()
+    }
+
+    override fun onResume() {
+        super.onResume()
+        setUniqueIdToView(listView)
+        if (activity is ScrollableViewInitializedListener) {
+            (activity as ScrollableViewInitializedListener).onScrollableViewInitialized(listView.id)
+        }
+
+        // Set the note if we retrieved the noteId from savedInstanceState
+        if (!TextUtils.isEmpty(restoredNoteId)) {
+            setNote(restoredNoteId)
+            reloadNoteBlocks()
+            restoredNoteId = null
+        }
+        if (notification == null) {
+            showErrorToastAndFinish()
+        }
+    }
+
+    override fun onPause() {
+        restoreOriginalViewId(listView)
+        super.onPause()
+    }
+
+    private fun setNote(noteId: String?) {
+        if (noteId == null) {
+            showErrorToastAndFinish()
+            return
+        }
+        val note: Note? = NotificationsTable.getNoteById(noteId)
```

Comment:
Good catch @jarvislin 👍 
I think it makes sense to put this in the background. I was thinking something like
```
    private fun setNote(noteId: String?, onNoteSet: (() -> Unit)? = null) {
        if (noteId == null) {
            showErrorToastAndFinish()
            return
        }
        lifecycleScope.launch(ioDispatcher) {
            val note: Note? = NotificationsTable.getNoteById(noteId)
            withContext(mainDispatcher) {
                if (note == null) {
                    showErrorToastAndFinish()
                } else {
                    notification = note
                }
                onNoteSet?.invoke()
            }
        }
    }
```
to callback when done and avoid any sync issue. I'm not sure if this would work in all cases (e.g. in the fragment `newInstance`) and I think we better cover it separately 🤔 
Note that [I duplicated this part of the code from the `NotificationsDetailListFragment`](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsDetailListFragment.kt#L174) and I was planning to sanitise this as the milestone implementation settles. 

> P.S. This is related to my Hack Week plan, I can handle it if you don't mind.

NP on this. Let me know how can I help 🙇